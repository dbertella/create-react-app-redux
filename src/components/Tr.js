import React, { PropTypes } from 'react';

const Tr = (props) => {
  const { info } = props;
  return (
    <tr>
      {
        info.map((artistInfo, i) => <td key={i}>{artistInfo}</td>)
      }
    </tr>
  );
};

Tr.propTypes = {
  info: PropTypes.array.isRequired,
}

export default Tr;
