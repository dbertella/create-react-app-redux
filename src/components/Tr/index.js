import React, { PropTypes } from 'react';

const Tr = (props) => {
  const { info } = props;
  return (
    <tr>
      {
        info.map((text, i) => <td key={i}>{text}</td>)
      }
    </tr>
  );
};

Tr.propTypes = {
  info: PropTypes.array.isRequired,
};

export default Tr;
