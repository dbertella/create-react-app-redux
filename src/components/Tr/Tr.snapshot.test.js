import React from 'react';
import renderer from 'react-test-renderer';

import Tr from './';

const artist = {
    "uuid": "f853578c-fc0f-4e65-81b8-566c5dffa35a",
    "gender": "F",
    "age": 39,
    "longitude": "-0.09998975",
    "rate": 14.21,
    "latitude": "51.75436293"
};

it('initial render', () => {
  const component = renderer.create(
    <Tr info={Object.keys(artist).map(key => artist[key])} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
