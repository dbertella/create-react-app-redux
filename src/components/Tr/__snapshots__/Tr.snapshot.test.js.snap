exports[`test initial render 1`] = `
<tr>
  <td>
    f853578c-fc0f-4e65-81b8-566c5dffa35a
  </td>
  <td>
    F
  </td>
  <td>
    39
  </td>
  <td>
    -0.09998975
  </td>
  <td>
    14.21
  </td>
  <td>
    51.75436293
  </td>
</tr>
`;
